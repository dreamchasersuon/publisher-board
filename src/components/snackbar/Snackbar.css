.Snackbar {
  position: fixed;
  bottom: 50px;
  right: 50px;
  padding: 20px;
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: 4px;
  background-color: white;
  box-shadow: 0 0 30px rgba(0, 0, 0, 0.25);
}

.error {
  color: #ebd300;
}

.success {
  color: #54be00;
}

.Snackbar-close {
  background-color: white;
  cursor: pointer;
  position: relative;
  left: 15px;
  bottom: 15px;
  color: black;
  width: 20px;
  height: 20px;
  border-radius: 50%;
  display: flex;
  align-items: center;
  justify-content: center;
}
