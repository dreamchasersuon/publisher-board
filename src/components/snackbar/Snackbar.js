import React from 'react';
import './Snackbar.css';

export default function Snackbar({ type, message, onClose }) {
  const classType = type === 'error' ? 'error' : 'success';
  return (
    <div className="Snackbar">
      <span className={'Snackbar-title ' + classType}>{message}</span>
      <div onClick={onClose} className="Snackbar-close">
        X
      </div>
    </div>
  );
}
