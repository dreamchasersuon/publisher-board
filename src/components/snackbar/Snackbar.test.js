import React from 'react';
import snapshotDiff from 'snapshot-diff';
import Snackbar from './Snackbar';

const handleOnPress = jest.fn();

test('renders error or success snack alert', () => {
  expect(
    snapshotDiff(
      <Snackbar
        onClose={handleOnPress}
        message="Something went wrong"
        type="error"
      />,
      <Snackbar
        onClose={handleOnPress}
        message="Operation succeed"
        type="success"
      />,
    ),
  ).toMatchSnapshot();
});
