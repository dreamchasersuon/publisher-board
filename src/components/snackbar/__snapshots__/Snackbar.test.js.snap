// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders error or success snack alert 1`] = `
"Snapshot Diff:
- <Snackbar message=\\"Something went wrong\\" onClose={[Function mockConstructor]} type=\\"error\\" />
+ <Snackbar message=\\"Operation succeed\\" onClose={[Function mockConstructor]} type=\\"success\\" />

@@ -1,12 +1,12 @@
  <div
    className=\\"Snackbar\\"
  >
    <span
-     className=\\"Snackbar-title error\\"
+     className=\\"Snackbar-title success\\"
    >
-     Something went wrong
+     Operation succeed
    </span>
    <div
      className=\\"Snackbar-close\\"
      onClick={[Function mockConstructor]}
    >"
`;
